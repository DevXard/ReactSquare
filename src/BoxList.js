import { useState } from 'react'
import NewBoxForm from './newBoxForm';
import Box from './Box';
import {v4 as uuid} from 'uuid'

const BoxList = () => {
    const [data, setData] = useState(
        []
    )
        console.log(data)
    const addNewSquare = (square) => {
        setData(data => ([...data, {...square, id:uuid()}]))
    }
    
    const removeSquare = (id) => {
        setData(data => data.filter(squ => squ.id !== id))
        
    }
    return (
        <div>
            <NewBoxForm addNewSquare={addNewSquare} />
            {data.map( square => <Box removeSquare={removeSquare} id={square.id} key={square.id} color={square.color} width={square.width} height={square.height} />)}
        </div>
    )
}

export default BoxList;
