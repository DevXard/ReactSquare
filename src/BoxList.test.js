import { render } from "@testing-library/react";
import BoxList from './BoxList';

it('should render without crashing', function() {
    render(<BoxList />)
})

it('should match snapshot', function() {
    const { asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot()
})

