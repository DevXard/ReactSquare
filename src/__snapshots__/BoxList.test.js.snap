// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=" "
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        type="number"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        type="number"
        value=""
      />
      <button>
        Add new color
      </button>
    </form>
  </div>
</DocumentFragment>
`;
